src/librustc_mir/interpret/eval_context.rs

```diff
@@ -721,6 +716,17 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         ::log_settings::settings().indentation -= 1;
         let frame = self.stack.pop().expect("tried to pop a stack frame, but there were none");
 
+        if !unwinding {
+            // Copy the return value to the caller's stack frame.
+            if let Some(return_place) = frame.return_place {
+                let op = self.access_local(&frame, mir::RETURN_PLACE, None)?;
+                self.copy_op(op, return_place)?;
```

Comment:
This should be `copy_op_transmute`; in general caller and callee return type can differ.

Reply:
Well, at least calling a method returning `usize` with a signature with `isize` return type should be fine (and `copy_op` would ICE on that). This check ensures that the return types are what Miri considers "function ABI compatible":

https://github.com/rust-lang/rust/blob/7fb5187d0423f4cd0441526571b8cd61927123c9/src/librustc_mir/interpret/terminator.rs#L353

Cc https://github.com/rust-lang/rust/issues/56166